es6 everywhere

jbellenger opened the pull request:
This updates g-w-p to use `class`, `let/const`, template literals, and arrow functions where appropriate. The [no-var](http://eslint.org/docs/rules/no-var) rule is added here to make the changes stick.

I also added an engines requirement to package.json:
```
"node": ">=4.3.0"
```
This value is inspired by [webpack's own engine requirement](https://github.com/webpack/webpack/blob/master/package.json#L72). Webpack appears to not accept node-5.0; I don't know what the history is there so I ignored it. 

**Testing done**:
- lint
- unit tests with node-4.3, node-4.8, node-6.10
- confirmed that dev and prod builds of twitter lite work

rxaviers commented on index.js:
```diff
@@ -11,16 +11,18 @@ var ProductionModePlugin = require("./ProductionModePlugin");
-function GlobalizePlugin(attributes) {
-  this.attributes = attributes || {};
-}
+class GlobalizePlugin {
```
On node `4.8`, I see the following sort of errors:

```
/Users/rxavier/globalizejs/globalize-webpack-plugin/index.js:14
class GlobalizePlugin {
^^^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
```

Adding ["use strict" declarations](https://gist.github.com/rxaviers/ccbf280c4fa7b07b0331f7b778081362) fixes such issues, though I still see the following error when running tests (even after the added use strict). I haven't looked any further... I noticed that on Node 6, this isn't a problem.

```
  Globalize Webpack Plugin
    Production Mode
      when using default module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          1) "before all" hook for "should extract formatters and parsers from basic code"
      when using named module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          2) "before all" hook for "should extract formatters and parsers from basic code"
      when using hashed module ids
        ✓ should extract formatters and parsers from basic code
        The compiled bundle
          3) "before all" hook for "should extract formatters and parsers from basic code"
```

jbellenger commented:
Thanks for testing this, Rafael! I (wrongly) assumed that it would just work. I tested this in node-4.3 and node-4.8 and saw that only 4.8 wanted "use strict". I've added it to all source files.

Your test failures look like they're coming from not having "use strict" in test/fixtures/app.js. I've added it there, as well as added an eslinter rule to error when use strict is not declared in a modules global scope.

rxaviers commented:
:tada: :tada: Thanks!
